atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionManager.java

```diff
@@ -415,6 +418,30 @@ <T, C extends PreCommitCondition, E extends Exception> T runTaskWithConditionRea
     @Timed
     long getUnreadableTimestamp();
 
+    /**
+     * Returns the timestamp that is before leased timestamps returned by the consumer on {@link TimestampLeaseAwareTransaction#preCommit(com.palantir.atlasdb.timelock.api.TimestampLeaseName, int, Consumer)}
+     * for a {@code timestampLeaseName} in open transactions.
+     * <p>
+     * This is similar to {@link #getImmutableTimestamp()} as it returns a timestamp before timestamps in open
+     * transactions,
+     * but for the immutable timestamp the timestamps considered are start timestamps for open transactions,
+     * while for leased timestamps the timestamps considered are leased timestamps from the corresponding
+     * {@code timestampLeaseName} in open transactions.
+     * <p>
+     * If no transactions with a {@code timestampLeaseName} lock are open, this method returns a new fresh timestamp
+     * (i.e. equivalent to {@link TimelockService#getFreshTimestamp()}).
+     * <p>
+     * Consumers should to fetch the leased timestamp outside of transactions that potentially use it - if fetching the
```

Comment:
I would say this is maybe a useful warning, but it assumes a lot about how this is used, therefore possibly making it not worth putting in here?